test/decomposer/test_zxz_decomposer.py

```diff
@@ -46,6 +46,20 @@ def test_zxz_decomposer(decomposer: ZXZDecomposer, gate: Gate, expected_result:
     assert decomposer.decompose(gate) == expected_result
 
 
+@pytest.mark.parametrize(
+    "gate",
+    [
+        Ry(0, -math.pi / 2),
+        BlochSphereRotation(0, axis=[0.0, 1.0, 0.0], angle=-math.pi / 2, phase=0.0),
+        BlochSphereRotation(0, axis=[0.0, -1.0, 0.0], angle=math.pi / 2, phase=0.0),
+    ],
+    ids=["Ry(-pi/2)", "BSR_1 of Ry(-pi/2)", "BSR_2 of Ry(-pi/2)"],
+)
+def test_specific_gate(decomposer: ZXZDecomposer, gate: Gate) -> None:
+    decomposed_gate = decomposer.decompose(gate)
+    check_gate_replacement(gate, decomposed_gate)
+
+
```

Comment:
This test was put here to reproduce the bug. Should we make a more general test that checks edge cases? It could be similar to the 'all  octant check' for the McKay decomposer, but that would also add a lot of testing time (the 'all octant check' already takes quite some time now...)

Reply:
The 'all octants check' is done for the ABA decomposer, Chris, i.e., for all XYX, XZX... ZYZ decomposers.

So we could get rid of this specific one, because it's covered in 'all octants check'.